analyses/molecular-subtyping-SHH-tp53/SHH-tp53-molecular-subtyping-data-prep.Rmd

```diff
@@ -0,0 +1,139 @@
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: Candace Savonen for ALSF CCDL
+date: 2019
+---
+
+This notebook [molecularly subtypes Medulloblastoma samples into their TP53 status](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/247). 
+
+## Usage
+
+This notebook is intended to be run via the command line from the top directory
+of the repository as follows:
+
+`Rscript -e "rmarkdown::render('analyses/molecular-subtyping-SHH-tp53/SHH-tp53-molecular-subtyping-data-prep.Rmd', clean = TRUE)"`
+
+## Set Up
+
+```{r}
+# Get `magrittr` pipe
+`%>%` <- dplyr::`%>%`
+```
+
+### Directories and Files
+
+```{r}
+# File path to data directory
+data_dir <- file.path("..", "..", "data")
+
+# File path to results directory
+results_dir <- "results"
+
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+## Read in the data
+
+Read in metadata and isolate to medulloblastoma samples.
+
+```{r}
+metadata <-
+  readr::read_tsv(file.path(data_dir, "pbta-histologies.tsv")) %>%
+  dplyr::filter(disease_type_new == "Medulloblastoma")
+
+# Check summary of subtypes here
+summary(as.factor(metadata$molecular_subtype))
+```
+
+Propagate the `molecular_subtype` labels to biospecimens of the same `sample_id`.
+First make a key of molecular subtype to `sample_id`.
+
+```{r}
+key_df <- metadata %>%
+  dplyr::filter(!is.na(molecular_subtype)) %>%
+  dplyr::select(molecular_subtype, sample_id)
+
+# Make into list form so it can be used in dplyr::recode
+key <- key_df$molecular_subtype
+names(key) <- key_df$sample_id
+```
+
+Recode and filter to all `sample_id`s that are `SHH` and have `WGS` or `WXS` data.
+
+```{r}
+shh_metadata <- metadata %>%
+  dplyr::mutate(molecular_subtype = dplyr::recode(sample_id, !!!key, .default = "NA")) %>%
+  dplyr::filter(molecular_subtype == "SHH")
+
+# Check summary after recoding
+summary(as.factor(shh_metadata$molecular_subtype))
+```
+
+Read in the SNV consensus calls. 
+
+```{r}
+snv_df <-
+  # Read in the consensus file
+  data.table::fread(file.path(data_dir, "pbta-snv-consensus-mutation.maf.tsv.gz"),
+    data.table = FALSE
+  ) %>%
+  # Filter to only Medulloblastoma samples
+  dplyr::filter(Tumor_Sample_Barcode %in% shh_metadata$Kids_First_Biospecimen_ID)
+```
+
+## Identify TP53 SNVs.
+
+Get a list of samples that are positive for TP53 mutations in the SNV consensus data.
+
+```{r}
+tp53_mut_biospecimens <- snv_df %>%
+  # Filter to only SNV's in TP53 gene
+  dplyr::filter(Hugo_Symbol == "TP53") %>%
+  dplyr::pull(Tumor_Sample_Barcode)
+```
+
+Now recode their TP53 status to be associated with their `sample_id`.
+
+```{r}
+# Make into list form so it can be used in dplyr::recode
+key <- shh_metadata$sample_id
+names(key) <- shh_metadata$Kids_First_Biospecimen_ID
+
+# List of tp53 postive samples
+tp53_mut_samples <- dplyr::recode(tp53_mut_biospecimens, !!!key)
+```
+
+Summarize TP53 status to a table with other sample identifiers.
+
+```{r}
+tp53_status_table <- shh_metadata %>%
```

Comment:
It does look like there are cases where a `sample_id` only has an RNA-seq sample and no WGS sample (e.g., `7316-1884`). I would say that that is _distinct_ from the case where you have a WGS sample and no mutation and should probably be a `NA` in `TP53_mutation`.

Reply:
Okay. I will add in a fix for this kind of case.